test/builders/wizards-v1/universe/universe-erc721-warper-builder-v1.ts

```diff
@@ -0,0 +1,131 @@
+      throw new SomethingNotConfiguredError();
+    }
+
+    this.universeConstructor = universeContructor;
+    return this;
+  }
+
+  public withWarperConstructor(warperConstructor: WarperConstructorV1): this {
+    if (this.warperConstructor) {
+      throw new SomethingNotConfiguredError();
+    }
+
+    this.warperConstructor = warperConstructor;
+    return this;
+  }
+
+  public withTaxTerms(taxTerms: ITaxTermsRegistry.TaxTermsStruct): this {
+    if (this.taxTerms) {
+      throw new SomethingNotConfiguredError();
+    }
+
+    this.taxTerms = taxTerms;
+    return this;
+  }
+
+  public withWarperAddress(address: string): this {
+    this.warperAddress = address;
+    return this;
+  }
+
+  public withWarperName(name: string): this {
+    this.warperName = name;
+    return this;
+  }
+
+  public async setupUniverseWithSuccess(): Promise<{ universeId: BigNumberish; deployedWarperAddress: string }> {
+    const { universeId, deployedWarperAddress } = await this.setupUniverseWithSuccessAndTx();
+    return { universeId, deployedWarperAddress };
+  }
+
+  public async setupUniverseWithSuccessAndTx(): Promise<{
+    universeId: BigNumberish;
+    deployedWarperAddress: string;
+    setupUniverseSuccessTx: ContractTransaction;
+  }> {
+    const [universeParams, taxTerms, warperAddress, warperParams, warperPresetId, warperInitData] =
+      this.prepareParams();
+
+    const { universeId, deployedWarperAddress } = await this.universeWizardV1.callStatic.setupUniverseAndWarper(
+      universeParams,
+      taxTerms,
+      warperAddress,
+      warperParams,
+      warperPresetId,
+      warperInitData,
+    );
+
+    return {
+      universeId,
+      deployedWarperAddress,
+      setupUniverseSuccessTx: await this.universeWizardV1.setupUniverseAndWarper(
+        universeParams,
+        taxTerms,
+        warperAddress,
+        warperParams,
+        warperPresetId,
+        warperInitData,
+      ),
+    };
+  }
+
+  public async setupUniverseWithTxPromiseOnly(): Promise<ContractTransaction> {
+    const [universeParams, taxTerms, warperAddress, warperParams, warperPresetId, warperInitData] =
+      this.prepareParams();
+    return this.universeWizardV1.setupUniverseAndWarper(
+      universeParams,
+      taxTerms,
+      warperAddress,
+      warperParams,
+      warperPresetId,
+      warperInitData,
+    );
+  }
+
+  private prepareParams(): [
+    IUniverseRegistry.UniverseParamsStruct,
+    ITaxTermsRegistry.TaxTermsStruct,
+    string,
+    IWarperManager.WarperRegistrationParamsStruct,
+    BytesLike,
+    BytesLike,
+  ] {
+    if (!this.universeConstructor || !this.warperConstructor || !this.taxTerms) {
+      throw new SomethingNotConfiguredError();
+    }
+
+    return [
+      this.universeConstructor.prepareUniverseParams(),
+      this.taxTerms,
+      this.warperAddress,
+      { name: this.warperName, universeId: 0, paused: false },
+      ERC721_WARPER_PRESET_IDS.ERC721_CONFIGURABLE_PRESET,
```

Comment:
Yes, and I realized that this will be a generic universe with warper builder. I do not know about `WarperRegistrationConstructorV1 ` - I looked at it, realized I do not want to use it because I think it is meant for a specific purpose. So that is why I created another one.